### DISCUSSION TITLE
How do you work with gc.cpp in IDE?

### DISCUSSION THREAD

**Discussion by huoyaoyuan** [dotnet/runtime#42898] [Q&A] [answered]
It's is included in `gcsvr.cpp` and `gcwks.cpp`, with required `.h` also in them. Visual Studio cannot see the `.h` included when browsing `gc.cpp`, and gets thousands of errors about missing declarations.

Since all the included headers are same, with only difference of `#define SERVER_GC 1`, can they be moved to `gc.cpp`?

@Maoni0

**Comment by Maoni0** [comment, marked as answer]
I don't use VS other than as an editor so I'm not familiar. I know that in VS you could do stuff like go to definition of a function in gc.cpp if you load up the project so it does work somewhat at least. 

CC @dotnet/gc maybe other folks on the team have an opinion.

**Reply by huoyaoyuan** [reply]
Thanks for your reply!
Navigation experience is quite important for new code readers. I guess that the structure of the file was from many years ago that you are very familiar with, but newcomers do need help.

**Reply by Maoni0** [reply]
actually it's always been this way since I started working on GC (this whole gcwks.cpp/gcsvr.cpp were done by someone else and way before my time). the way I learned the code base was by running a simple test under a debugger and stepping through the GC functions. the BotR GC chapter lists [the top level functions](https://github.com/dotnet/runtime/blob/master/docs/design/coreclr/botr/garbage-collection.md#physical-architecture). I would suggest to set breakpoints on them and step through the code that way.  let us know if you have issues running a test.
